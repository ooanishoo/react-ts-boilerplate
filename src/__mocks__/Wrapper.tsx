import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'

/* a wrapper Router component that takes any React component and renders as it's child component */
type WrapperProps = {
    component?: React.ReactNode
}
export const Wrapper: React.FC<WrapperProps> = ({ component }: WrapperProps) => <Router>{component}</Router>
