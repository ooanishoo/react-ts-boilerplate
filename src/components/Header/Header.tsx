import React from 'react'
import { Link } from 'react-router-dom'

const Header: React.FC = () => {
    return (
        <header data-testid="header">
            <h1>My Project</h1>
            <nav data-testid="navigation" role="navigation">
                <ul>
                    <li>
                        <Link to="/">Home</Link>
                    </li>
                    <li>
                        <Link to="/todos">Todos</Link>
                    </li>
                </ul>
            </nav>
        </header>
    )
}

export default Header
