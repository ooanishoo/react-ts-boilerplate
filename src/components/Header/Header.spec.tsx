import { render, within } from '@testing-library/react'
import React from 'react'
import Header from './Header'
import { Wrapper } from '../../__mocks__/Wrapper'

describe('<Header/>', () => {
    test('should render without crashing', () => {
        const { container, getByTestId, queryByRole } = render(<Wrapper component={<Header />} />)
        expect(container).toMatchSnapshot()

        const header = getByTestId('header')
        expect(header).toBeInTheDocument()

        const navigation = within(header).getAllByTestId('navigation')
        expect(navigation.length).toBe(1)
        expect(queryByRole('navigation')).toBeTruthy()
    })
})
