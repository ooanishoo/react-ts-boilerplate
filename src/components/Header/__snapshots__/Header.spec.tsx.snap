// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should render without crashing 1`] = `
<div>
  <header
    data-testid="header"
  >
    <h1>
      My Project
    </h1>
    <nav
      data-testid="navigation"
      role="navigation"
    >
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/todos"
          >
            Todos
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
